Created Confusion matrix function as well as unit test functions

miketianchen opened the pull request:
should solve #20

miketianchen committed:
completed confusion matrix function

miketianchen committed:
Confusion_matrix function is written and tested for

anasm-17 commented on R/confusion_matrix.R:
```diff
@@ -2,20 +2,66 @@
 #' visual. If predicted_y array is passed in,other evaluation scoring metrics such as
 #' Recall, and precision will also be produced
```
Does it only work for binary classification? if so it is worth mentioning the limitations

miketianchen replied:
![image](https://user-images.githubusercontent.com/38637744/76153255-57003a80-607e-11ea-9c81-e1bb181a3297.png)

works for more than 2 classes

anasm-17 commented on R/confusion_matrix.R:
```diff
@@ -2,20 +2,66 @@
+  metric_score <- broom::tidy(confusion_matrix$byClass)
+
+  print(confusion_plot)
+  return(metric_score)
```
I think it maybe worth returning the table as well.
```suggestion
  return(list(cf_table = table, metric = metric_score))
```

miketianchen replied:
![image](https://user-images.githubusercontent.com/38637744/76153222-0688dd00-607e-11ea-900b-a0e548261834.png)


that is the table.

anasm-17 reviewed (approved):
See changes for future improvement :) looks good otherwise

anasm-17 committed:
Fixed merge conflict

Merge branch 'master' into confusion_dev

# Conflicts:
#	DESCRIPTION
#	NAMESPACE
#	man/confusion_matrix.Rd
#	man/modelcomptab.Rd
#	man/plot_train_valid_acc.Rd

anasm-17 committed:
Merge branch 'master' into confusion_dev
